acclimatise/model.py

```diff
@@ -371,33 +371,62 @@ def capital(self):
         )
 
 
-int_re = re.compile("(int(eger)?)|size|length|max|min", flags=re.IGNORECASE)
-str_re = re.compile("str(ing)?", flags=re.IGNORECASE)
-float_re = re.compile("float|decimal", flags=re.IGNORECASE)
-bool_re = re.compile("bool(ean)?", flags=re.IGNORECASE)
-file_re = re.compile("file|path", flags=re.IGNORECASE)
-dir_re = re.compile("folder|directory", flags=re.IGNORECASE)
+int_re = re.compile(
+    r"\b((int(eger)?)|size|length|max|min|(num(ber)?))\b", flags=re.IGNORECASE
+)
+str_re = re.compile(r"\bstr(ing)?\b", flags=re.IGNORECASE)
+float_re = re.compile(r"\b(float|decimal)\b", flags=re.IGNORECASE)
+bool_re = re.compile(r"\bbool(ean)?\b", flags=re.IGNORECASE)
+file_re = re.compile(r"\b(file|path)\b", flags=re.IGNORECASE)
+input_re = re.compile(r"input", flags=re.IGNORECASE)
+output_re = re.compile(r"output", flags=re.IGNORECASE)
+dir_re = re.compile(r"\b(folder|directory)\b", flags=re.IGNORECASE)
+
+default_re = re.compile(
+    r"default(?: value)?(?:[:=] ?| )([^ )\]]+)", flags=re.IGNORECASE
+)
+float_num_re = re.compile(
+    r"[+-]?(([0-9]*\.[0-9]+)|((?:0|[1-9]\d*)(?:\.\d*)?(?:[eE][+\-]?\d+)))",
+    flags=re.IGNORECASE,
+)
+int_num_re = re.compile(r"([+-]?[0-9]+)", flags=re.IGNORECASE)
 
 
 def infer_type(string) -> typing.Optional[cli_types.CliType]:
     """
     Reads a string (argument description etc) to find hints about what type this argument might be. This is
     generally called by the get_type() methods
     """
-    if bool_re.match(string):
```

Comment:
Nice pickup, I suppose this never worked properly if I was using `match` before.

Reply:
Yes, for many calls it did not work... I guess.